src/main/java/io/confluent/connect/hdfs/wal/WALFile.java

```diff
@@ -97,6 +97,7 @@ private static int getBufferSize(Configuration conf) {
     private FSDataOutputStream out;
     private DataOutputBuffer buffer = new DataOutputBuffer();
     private boolean appendMode;
+    private FileSystem fs = null;
```

Comment:
Also, can this field be `final` since it is only set in the constructor?

Reply:
I'm not sure how -- I can't create it without the "Options" passed into the constructor. Maybe there's a method I'm not aware of?

Reply:
I'm having trouble getting this to work since the "catch" block needs `fs` to be defined. I think this will be a battle we can fight after we get feedback from a field deployment that this patch does in fact fix the issue

Reply:
Yeah, that's fine. It's not worth that much effort and trouble. Plus the other variables are not defined as final, either.